### PR TITLE
Remove lookarounds from autolink extension patterns

### PR DESCRIPTION
**Pull request by chenzhiguang**
Fix: https://github.com/dart-lang/markdown/issues/515

### CODE REVIEW

**Comment by chenzhiguang** [comment]
There might be some other places using RegExp lookarounds, and I will fix them if there are any.

File: lib/src/inline_syntaxes/autolink_extension_syntax.dart [hunk 1]
```diff
@@ -13,7 +13,8 @@ class AutolinkExtensionSyntax extends InlineSyntax {
   static const _linkPattern =
       // Autolinks can only come at the beginning of a line, after whitespace,
       // or any of the delimiting characters *, _, ~, and (.
-      r'(?<=^|[\s*_~(>])'
+      // Note: Disable it, as Safari does not support lookarounds.
```
Comment (srawlins) [inline lib/src/inline_syntaxes/autolink_extension_syntax.dart:16]: For future archaeologists, we can expand on this comment a bit, like "Disable this piece for now, as... Consider re-enabling later."

File: lib/src/inline_syntaxes/autolink_extension_syntax.dart [hunk 2]
```diff
@@ -55,6 +58,24 @@ class AutolinkExtensionSyntax extends InlineSyntax {
...
+      final precededBy = String.fromCharCode(parser.charAt(parser.pos - 1));
+      if ([' ', '*', '_', '~', '(', '>'].contains(precededBy) == false) {
+        return false;
+      }
+    }
+
+    // When it is an email link and followed by `_` or `-`, it is invalid.
+    if (startMatch[2] != null && parser.source.length > startMatch.end) {
+      final followedBy = String.fromCharCode(parser.charAt(startMatch.end));
+      if (['_', '-'].contains(followedBy)) {
```
Comment (srawlins) [inline lib/src/inline_syntaxes/autolink_extension_syntax.dart:74]: This can be a const set as well.

**Review by srawlins — CHANGES_REQUESTED** [review]
Thanks for the quick fix!!

File: lib/src/inline_syntaxes/autolink_extension_syntax.dart (hunk 1 above)
Comment (chenzhiguang) [reply lib/src/inline_syntaxes/autolink_extension_syntax.dart]: Fixed!

File: lib/src/inline_syntaxes/autolink_extension_syntax.dart [hunk 3]
```diff
@@ -55,6 +58,24 @@ class AutolinkExtensionSyntax extends InlineSyntax {
     if (startMatch == null) {
       return false;
     }
+
+    // When it is a link and it is not preceded by `*`, `_`, `~`, `(`, or `>`,
+    // it is invalid.
```
[the remark this answers is not in the document]
Comment (chenzhiguang) [reply lib/src/inline_syntaxes/autolink_extension_syntax.dart]: Fixed!

File: lib/src/inline_syntaxes/autolink_extension_syntax.dart [hunk 4]
```diff
@@ -55,6 +58,24 @@ class AutolinkExtensionSyntax extends InlineSyntax {
...
+    if (startMatch[1] != null && parser.pos > 0) {
+      final precededBy = String.fromCharCode(parser.charAt(parser.pos - 1));
+      if ([' ', '*', '_', '~', '(', '>'].contains(precededBy) == false) {
```
[the remark this answers is not in the document]
Comment (chenzhiguang) [reply lib/src/inline_syntaxes/autolink_extension_syntax.dart]: Fixed!

File: lib/src/inline_syntaxes/autolink_extension_syntax.dart (hunk 2 above)
Comment (chenzhiguang) [reply lib/src/inline_syntaxes/autolink_extension_syntax.dart]: Fixed!

**Comment by kevmoo** [comment]
`dart test -p safari` now passes with these changes! thanks!

**Review by srawlins — APPROVED** [review]
❤️  Thanks!

**Comment by chenzhiguang** [comment]
Nice, thank you too!

**Pull request merged**